packages/dds/matrix/src/test/time/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,134 @@
+					benchmarkFn: () => {
+						for (let i = 0; i < count; i++) {
+							localMatrix.insertCols(Math.floor(localMatrix.colCount / 2), 1);
+						}
+					},
+				});
+
+				// Test the execute time of the SharedMatrix for inserting a row in the middle for a given number of times.
+				benchmark({
+					title: `Insert a row in the middle ${count} times`,
+					benchmarkFn: () => {
+						for (let i = 0; i < count; i++) {
+							localMatrix.insertRows(Math.floor(localMatrix.rowCount / 2), 1);
+						}
+					},
+				});
+
+				// Test the execute time of the SharedMatrix for inserting a row and a column in the middle for a given number of times.
+				benchmark({
+					title: `Insert a row and a column ${count} times`,
+					benchmarkFn: () => {
+						for (let i = 0; i < count; i++) {
+							localMatrix.insertCols(Math.floor(localMatrix.colCount / 2), 1);
+							localMatrix.insertRows(Math.floor(localMatrix.rowCount / 2), 1);
+						}
+					},
+				});
+			}
+
+			// Set/Remove-related tests that are limited by matrixSize
+			for (const count of validRemoveCounts) {
+				// Test the execute time of the SharedMatrix for removing a column in the middle for a given number of times.
+				benchmark({
+					title: `Remove the middle column ${count} times`,
+					benchmarkFn: () => {
+						localMatrix = createLocalMatrix({
+							id: "testLocalMatrix",
+							size: matrixSize,
+							initialValue: matrixValue,
+						});
+						for (let i = 0; i < count; i++) {
+							localMatrix.removeCols(Math.floor(localMatrix.colCount / 2), 1);
+						}
+					},
+				});
+
+				// Test the execute time of the SharedMatrix for removing a row in the middle for a given number of times.
+				benchmark({
+					title: `Remove the middle row ${count} times`,
+					benchmarkFn: () => {
+						localMatrix = createLocalMatrix({
+							id: "testLocalMatrix",
+							size: matrixSize,
+							initialValue: matrixValue,
+						});
+						for (let i = 0; i < count; i++) {
+							localMatrix.removeRows(Math.floor(localMatrix.rowCount / 2), 1);
+						}
+					},
+				});
+
+				// Test the execute time of the SharedMatrix for removing a row and a column in the middle for a given number of times.
+				benchmark({
+					title: `Remove the middle row and column ${count} times`,
+					benchmarkFn: () => {
+						localMatrix = createLocalMatrix({
+							id: "testLocalMatrix",
+							size: matrixSize,
+							initialValue: matrixValue,
+						});
+						for (let i = 0; i < count; i++) {
+							localMatrix.removeCols(Math.floor(localMatrix.colCount / 2), 1);
+							localMatrix.removeRows(Math.floor(localMatrix.rowCount / 2), 1);
+						}
+					},
+				});
+
+				// Test the execute time of the SharedMatrix for setting a string in a cell for a given number of times.
+				benchmark({
+					title: `Set a 3-character string in ${count} cells`,
+					benchmarkFn: () => {
+						for (let i = 0; i < count; i++) {
+							localMatrix.setCell(i, i, "abc");
```

Comment:
Shouldn't we be careful and create a new local matrix *just* for this set-test? Seems like we are using using `localMatrix` from the latest remove benchmark test (the one right above). 

After running the remove-test on `localMatrix` of size `10x10` for `10` times, don't we then have an empty matrix? I was expecting `setCell(0, 0)` to throw an error due to index out of range, but the benchmark test seems to run just fine. 

Thoughts? @alexvy86 @chentong7

Reply:
This is within the same test suite as the tests above so `beforeEach()` is creating a new matrix (right now three times 😄, see one of my other comment) for each test, including this one.

Reply:
Updated and tested. Resolving this one...